### PR TITLE
Update ROADMAP.md

### PR DESCRIPTION
**Pull request by ttmc**
I updated the ROADMAP.md file (in the root directory of the repository).

Mostly, I moved stuff that is now done to a new "Done/Past" section at the bottom. I realize it's rather silly to have such a section in the ROADMAP.md file, but we haven't settled on a way to record release notes yet (e.g. Use GitHub "Releases" tags, use a CHANGELOG.md file, both, or what).

I also added some stuff-to-do. Did I miss something?

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -2,20 +2,27 @@
 
 ## BigchainDB Protocols
 * Validation of other nodes
-* Byzantine fault tolerance
```
Comment (r-marques) [inline ROADMAP.md]: We should leave at least fault tolerance

Comment (ttmc) [inline ROADMAP.md]: Okay I pushed that change to this branch.

**Comment by r-marques** [comment]
:+1:

**Comment by vrde** [comment]
@ttmc consider adding the items in the [API Roadmap](https://github.com/bigchaindb/bigchaindb/pull/102/files#diff-87103e340158c4591f965b8cda8a9207R44) as well.

**Comment by ttmc** [comment]
@vrde I have a TODO to refactor the API documentation (because there are many things called "API", e.g. HTTP API, Python API, etc.). I think I will move that roadmap stuff out to `ROADMAP.md` when I do that, and maybe link to `ROADMAP.md`.

**Pull request merged**